pkg/audit/audit.go

```diff
@@ -27,8 +35,92 @@ func (d *LoggingAudit) Write(q *QueryData) error {
 	return nil
 }
 
-type CloudwatchAudit struct{}
+type SplunkEventData struct {
+	Query     string `json:"query"`
+	User      string `json:"user"`
+	Namespace string `json:"namespace"`
+	Pod       string `json:"pod"`
+}
 
-func (c *CloudwatchAudit) Write(q *QueryData) error {
-	return nil
+type SplunkQueryData struct {
+	Event      *SplunkEventData `json:"event"`
+	Time       int64 `json:"time"`
+	Host       string `json:"host"`
+	Source     string `json:"source"`
+	Sourcetype string `json:"sourcetype"`
+	Index      string `json:"index"`
 }
+
+type SplunkResponse struct {
+	Text string
+	Code int
+}
+
+type SplunkAudit struct {
+	Index      string
+	Token      string
+	Endpoint   string
+	Host       string
+	Pod        string
+	Namespace  string
+	Source     string
+	Sourcetype string
+}
+
+func (d *SplunkAudit) Init(se *splunk.Splunkenv) {
```

Comment:
Passing these variables around like this feels a bit awkward. I am thinking on how to simplify and I'm open to any ideas.

Reply:
I think you can just define 
```
type SplunkAudit struct {
  env Splunkenv
}
```

Reply:
I think you can just define 
```
type SplunkAudit struct {
  env Splunkenv
}
```